backend/src/controllers/filter-item-controller.ts

```diff
@@ -0,0 +1,152 @@
+import { Request, Response } from 'express'; 
+import { validationResult } from 'express-validator'; 
+import InventoryItem from '../models/inventory-item'; 
+import Item from '../models/item';
+import mongoose from 'mongoose';
+
+interface ItemInterface {
+    _id: string,
+    name: string,
+    description: string,
+    pickupLocation: string,
+    pickupTime: string
+}
+
+export const getFilteredInventoryItems = async (req: Request, res: Response) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+        console.log("errors", errors);
+        return res.status(422).json({ errors: errors.array() });
+    }
+
+    try {
+        // Filter
+        let fetchedInventoryItems: any = {};
+        const itemOwner = req.query.itemOwner as string
+        const size = req.query.size as string
+        const name = req.query.name as string
+
+        // Sort
+
+        const sortedBy = req.query.sortedBy as string
+
+        if (itemOwner && size && name) {
```

Comment:
I have some suggestions on how to simplify the sort logic; let's cover them on the call.